codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -434,9 +453,10 @@ public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data) throws Http2E
         @Override
         public void onPushPromiseRead(ChannelHandlerContext ctx, int streamId, int promisedStreamId,
                 Http2Headers headers, int padding) throws Http2Exception {
-            Http2Stream parentStream = connection.requireStream(streamId);
+            Http2Stream parentStream = connection.stream(streamId);
 
-            if (streamCreatedAfterGoAwaySent(parentStream)) {
+            if (parentStream == null || streamCreatedAfterGoAwaySent(streamId)) {
```

Comment:
I think you got it [section6.6](https://tools.ietf.org/html/draft-ietf-httpbis-http2-17#section-6.6):

>Since PUSH_PROMISE reserves a stream, ignoring a PUSH_PROMISE frame
   causes the stream state to become indeterminate.  A receiver MUST
   treat the receipt of a PUSH_PROMISE on a stream that is neither
   "open" nor "half closed (local)" as a connection error
   (Section 5.4.1) of type PROTOCOL_ERROR.